src/Agent.Worker/Release/ReleaseJobExtension.cs

```diff
@@ -0,0 +1,258 @@
+                ArgUtil.NotNullOrEmpty(agentArtifactDefinition.Alias, nameof(agentArtifactDefinition.Alias));
+
+                executionContext.Output(StringUtil.Loc("RMStartArtifactsDownload"));
+                ArtifactDefinition artifactDefinition = ConvertToArtifactDefinition(agentArtifactDefinition, executionContext);
+                executionContext.Output(StringUtil.Loc("RMArtifactDownloadBegin", agentArtifactDefinition.Alias));
+                executionContext.Output(StringUtil.Loc("RMDownloadArtifactType", agentArtifactDefinition.ArtifactType));
+
+                // Get the local path where this artifact should be downloaded. 
+                string downloadFolderPath = GetLocalDownloadFolderPath(agentArtifactDefinition, artifactsWorkingFolder);
+
+                // Create the directory if it does not exist. 
+                if (!Directory.Exists(downloadFolderPath))
+                {
+                    // TODO: old windows agent has a directory cache, verify and implement it if its required.
+                    Directory.CreateDirectory(downloadFolderPath);
+                    executionContext.Output(StringUtil.Loc("RMArtifactFolderCreated", downloadFolderPath));
+                }
+
+                // download the artifact to this path. 
+                RetryExecutor retryExecutor = new RetryExecutor();
+                await retryExecutor.ExecuteAsync(
+                    async () =>
+                        {
+                            //TODO:SetAttributesToNormal
+                            var releaseFileSystemManager = HostContext.GetService<IReleaseFileSystemManager>();
+                            releaseFileSystemManager.CleanupDirectory(downloadFolderPath);
+
+                            if (agentArtifactDefinition.ArtifactType == AgentArtifactType.GitHub
+                                || agentArtifactDefinition.ArtifactType == AgentArtifactType.TFGit
+                                || agentArtifactDefinition.ArtifactType == AgentArtifactType.Tfvc)
+                            {
+                                throw new NotImplementedException();
+                            }
+                            else
+                            {
+                                var buildArtifactProvider = HostContext.GetService<IArtifactProvider>();
+                                await buildArtifactProvider.Download(executionContext, artifactDefinition, downloadFolderPath);
+                            }
+                        });
+
+                executionContext.Output(StringUtil.Loc("RMArtifactDownloadFinished", agentArtifactDefinition.Alias));
+            }
+
+            executionContext.Output(StringUtil.Loc("RMArtifactsDownloadFinished"));
+        }
+
+        private static string GetLocalDownloadFolderPath(AgentArtifactDefinition artifactDefinition, string artifactsWorkingFolder)
```

Comment:
right.. it had other changes.. I did some cleanup before I moved the code to the new agent and it became just onliner.. thanks for pointing out. Fixed it.